holmes/core/investigation_structured_output.py

```diff
@@ -67,6 +68,17 @@ def process_response_into_sections(response: Any) -> Tuple[str, OutputSectionsDa
         # if it's not a string, we make it so as it'll be parsed later
         response = str(response)
 
+    if response.startswith("```json\n") and response.endswith("\n```"):
+        try:
+            parsed = json.loads(
+                response[8:-3]
+            )  # if this parses as json, set the response as that.
+            if isinstance(parsed, dict):
+                logging.warning("LLM did not return structured data")
+                response = response[8:-3]
```

Comment:
do we want to set the response to the `parsed` or to the `response`?
Also, I think we might want to return, because in this case the json parsing on line 83 will always fail

Reply:
This code is removing the leading ` ```json ` and trailing ` ``` ` from a string.
This then gets parsed again further below (current line is 156). I'm doing this double parsing because I think it "simplifies" reading and understanding it but I could be wrong. Happy to change if it's confusing.